mobile/src/user/UserPages.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { describe, it, expect, beforeEach, afterEach, vi } from "vitest";
+import { mock } from "vitest-mock-extended";
+
+import { cleanup, render, screen, fireEvent } from "@testing-library/react";
+import matchers from "@testing-library/jest-dom/matchers";
+
+expect.extend(matchers);
+
+let setUser: any;
+let userRepository: any;
+let navigate: any;
+
+vi.mock("../shared/SharedHooks", () => {
+  return {
+    useUser: function () {
+      return [null, setUser];
+    },
+    useUserRepository: function () {
+      return userRepository;
+    },
+  };
+});
+
+vi.mock("react-router-dom", () => {
+  return {
+    useNavigate: function () {
+      return navigate;
+    },
+  };
+});
+
+beforeEach(() => {
+  setUser = vi.fn();
+  userRepository = mock<UserRepository>();
+  navigate = vi.fn();
+});
+
+afterEach(() => {
+  cleanup();
+  vi.clearAllMocks();
+});
+
+import { SignIn, SignOut } from "./UserPages";
+import { UserRepository } from "./UserRepository";
+
+describe("SignIn", () => {
+  const testName = "test-user";
+  const testUser = { id: 1, name: testName };
+
+  it("Should sign in user if name is set", async () => {
+    userRepository.signIn.mockResolvedValueOnce(testUser);
+
+    render(<SignIn />);
+
+    const nameField = await screen.findByLabelText("Username");
+    const button = await screen.findByRole("button");
+
+    fireEvent.change(nameField, {
+      target: { value: testName },
+    });
+
+    fireEvent.click(button);
+
+    await screen.findAllByText("");
+
+    expect(userRepository.signIn).toBeCalledWith(testName);
```

Comment:
You're testing implementation details. This test will break when you refactor, even though the app still works. Instead, you should test that the feature works from the user point of view (in that case, that the user sees their name after sign in). 

https://kentcdodds.com/blog/testing-implementation-details